packages/foam-core/src/markdown-provider.ts

```diff
@@ -143,45 +157,78 @@ export function createMarkdownReferences(
     return [];
   }
 
-  return graph
-    .getForwardLinks(noteId)
-    .map(link => {
-      let target = graph.getNote(link.to);
-      // if we don't find the target by ID we search the graph by slug
-      if (!target) {
-        const candidates = graph.getNotes({ slug: link.link.slug });
-        if (candidates.length > 1) {
+  // if note doesn't exist, we can't find its links
+  const note = graph.getNote(noteId);
+  if (!note) {
+    return [];
+  }
+
+  const forwardLinks = graph.getForwardLinks(noteId);
+
+  // Try to generate a definition for each [[link]] in the note.
+  //
+  // A note may have multiple [[link]] expressions to the same target
+  // note with different aliases.s
+  //
+  // - note.links contains all the [[link]] expressions
+  // - forwardLinks contains all the edges (links) between documents
+  return (
+    note.links
+      .map(linkExpression => {
+        // find the link between this and other document
+        const link = forwardLinks.find(
+          note => note.link.slug === linkExpression.slug
```

Comment:
I was finding this hard to understand, then I think this parameter is not really `note`, is it? isn't it the graph connection?

This might partially be a naming issue (as connection/link has a `link` attribute, which makes things harder to read), not sure what's the best naming pattern we should adopt (maybe `connection`?)

Reply:
Hmm... yeah, this is confusing to me too now. I'll look into it once I loop back to this PR -- for now I'm not sure if we should move ahead with this until we address the GitHub Pages rendering issues.